x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -347,13 +321,369 @@ public void testDataStreams_autoFollowAfterDataStreamCreated() throws Exception
+                ensureYellow(aliasName);
+                verifyDocuments(client(), aliasName, numDocs);
+            });
+        }
+
+        // Rollover in leader cluster and ensure second backing index is replicated:
+        {
+            try (RestClient leaderClient = buildLeaderClient()) {
+                Request rolloverRequest = new Request("POST", "/" +  aliasName + "/_rollover");
+                assertOK(leaderClient.performRequest(rolloverRequest));
+                verifyAlias(leaderClient, aliasName, true, aliasName + "-000002", aliasName + "-000001");
+
+                Request indexRequest = new Request("POST", "/" + aliasName + "/_doc");
+                indexRequest.addParameter("refresh", "true");
+                indexRequest.setJsonEntity("{\"@timestamp\": \"" + DATE_FORMAT.format(new Date()) + "\",\"message\":\"abc\"}");
+                assertOK(leaderClient.performRequest(indexRequest));
+                verifyDocuments(leaderClient, aliasName, numDocs + 1);
+            }
+            assertBusy(() -> {
+                assertThat(getNumberOfSuccessfulFollowedIndices(), equalTo(initialNumberOfSuccessfulFollowedIndices + 2));
+                verifyAlias(client(), aliasName, false, aliasName + "-000002", aliasName + "-000001");
+                ensureYellow(aliasName);
+                verifyDocuments(client(), aliasName, numDocs + 1);
+            });
+        }
+
+        // Try rollover in follow cluster, this should fail, because is_write_index property of an alias isn't
+        // replicated to follow cluster.
+        {
+            Request rolloverRequest1 = new Request("POST", "/" +  aliasName + "/_rollover");
+            Exception e = expectThrows(ResponseException.class, () -> client().performRequest(rolloverRequest1));
+            assertThat(e.getMessage(), containsString("rollover target [" + aliasName + "] does not point to a write index"));
+            verifyAlias(client(), aliasName, false, aliasName + "-000002", aliasName + "-000001");
+        }
+        // Cleanup:
+        {
+            deleteAutoFollowPattern("test_pattern");
+        }
+    }
+
+    private static void verifyAlias(RestClient client,
+                                    String aliasName,
+                                    boolean checkWriteIndex,
+                                    String... otherIndices) throws IOException {
+        Request getAliasRequest = new Request("GET", "/_alias/" + aliasName);
+        Map<?, ?> responseBody = toMap(client.performRequest(getAliasRequest));
+        if (checkWriteIndex) {
+            assertThat(ObjectPath.eval(otherIndices[0] + ".aliases." + aliasName + ".is_write_index", responseBody), is(true));
+        }
+        for (String otherIndex : otherIndices) {
+            assertThat(ObjectPath.eval(otherIndex + ".aliases." + aliasName, responseBody), notNullValue());
+        }
+    }
+
+    public void testDataStreamsBiDirectionalReplication() throws Exception {
```

Comment:
@martijnvg Sorry, I think I didn't explain well my concern about supporting bi-directional replication. What I meant was JasonZ's [blog](https://www.elastic.co/blog/bi-directional-replication-with-elasticsearch-cross-cluster-replication-ccr). In his setup, we can send the same indexing request (uses the same write alias) to either cluster. In this test, we use different data streams for indexing requests. That means users can't simply reroute all indexing to a single cluster when another is not available.

Reply:
Currently an alias can't point to a data stream or its backing indices. So the test can't completely follow the bi directional scenario. I did add a few `logs-http*` searches to mimic reading from a `logs-http` alias, as best effort replacement for the fact aliases can't point to data streams.

We're planning to add alias support to data streams. These aliases would only be able point to data streams and not to a data stream's backing indices, other indices or other aliases. Like aliases defined on indices, aliases on data stream could also have a write flag, which indicates to which data stream write requests are redirected to. I will add a TODO here, that these searches on `logs-http*` pattern should be replaced with searches and writes via aliases when alias support for data streams has landed.

Reply:
> We're planning to add alias support to data streams.

Thanks for explaning + adding the TODO.

Reply:
@dnhatn I've opened: #66163

Reply:
Thanks @martijnvg.